pkg/mesh/td.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mesh
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"strconv"
+	"strings"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/google/uuid"
+)
+
+// TDSidecarEnv contains environment files that controls how an envoy proxy will be
+// set up and interact with Traffic Director control plane.
+type TdSidecarEnv struct {
+	// Project Number where Traffic Director resources are configured.
+	ProjectNumber string
```

Comment:
It would be great to not duplicate - for example ProjectNumber is common.
NetworkName can also move to the other struct, we'll use it in ASM too. 

EnvoyPort/admin port - I don't think we want to expose them to users or make them configurable, it may be simpler to keep them as constant.

Same for most others - for example XdsServerCert, etc.

We can probably do this in separate iterations - but please at least add some TODOs and comments.

Reply:
Ack, I'll move/reuse ProjectNumber and NetworkName to the krun struct. I've also removed configurability for everything except projectNumber and NetworkName for TD.

Reply:
ProjectNumber and NetworkName are now used from KRun.